tricity/tricity/pipelines.py

```diff
@@ -1,13 +1,39 @@
-# Define your item pipelines here
-#
-# Don't forget to add your pipeline to the ITEM_PIPELINES setting
-# See: https://docs.scrapy.org/en/latest/topics/item-pipeline.html
+import pymongo
+from tricity import settings
+from datetime import datetime
+import hashlib
 
+ 
+class MongoDBPipeline:
+    
+    def set_hash(self,item):
+        result = hashlib.md5(f"{item['price']}{item['url']}".encode('utf-8'))
+        hash_value = result.hexdigest()
+        return hash_value
 
-# useful for handling different item types with a single interface
-from itemadapter import ItemAdapter
+    def set_scrapping_date(self):
+        now = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
+        return now
 
+    def open_spider(self, spider):
+        self.scrapping_date = self.set_scrapping_date()
+        self.client = pymongo.MongoClient(settings.mongodb_uri)
+        db = self.client[settings.mongodb_db]
+        self.collection = db[settings.colection_name]
+        #start with a clean database
+        self.collection.delete_many({})
+
+    def close_spider(self, spider):
+        self.client.close()
 
-class TricityPipeline:
     def process_item(self, item, spider):
-        return item
+        item['price_per_meter'] = str(round(float(item['price']) / float(item['area']), 2))
+        item['hash'] = self.set_hash(item)
+        item['scrapping_date'] = self.scrapping_date
+        data = dict(item)
+        if not item['hash'] in data:
```

Comment:
Musisz sprawdzić w bazie danych czy jest jakis  obiekt z tym hashem,
https://stackoverflow.com/questions/8389811/how-to-query-mongodb-to-test-if-an-item-exists
Jeżeli nie istnieje to robisz insert do bazy, jeżeli istnieje to zrobisz update w bazie.
https://www.mongodb.com/docs/mongodb-shell/crud/update/

Reply:
ok